android/app/src/main/java/com/breez/client/plugins/breez/breezlib/GoogleAuthenticator.java

```diff
@@ -59,28 +64,66 @@ public GoogleSignInAccount ensureSignedIn(final boolean silent) throws Exception
         }
     }
 
-    public String getAccessToken() throws Exception{
+    public String getAccessToken() throws Exception {
+        Log.d(TAG, "getAccessToken");
         GoogleSignInAccount googleAccount = ensureSignedIn(true);
-        GoogleAccountCredential credential =
-                GoogleAccountCredential.usingOAuth2(
-                        activityBinding.getActivity(), Collections.singleton(DriveScopes.DRIVE_APPDATA));
 
-        credential.setSelectedAccount(googleAccount.getAccount());
-        return credential.getToken();
+        GoogleAccountCredential credential;
+        try {
+            credential = GoogleAccountCredential.usingOAuth2(
+                    activityBinding.getActivity(),
+                    Collections.singleton(DriveScopes.DRIVE_APPDATA));
+        } catch (Exception e) {
+            Log.w(TAG, "getAccessToken failed", e);
+            throw e;
+        }
+
+        try {
+            credential.setSelectedAccount(googleAccount.getAccount());
+        } catch (Exception e) {
+            Log.w(TAG, "setSelectedAccount failed", e);
+            throw e;
+        }
+
+        try {
+            return credential.getToken();
+        } catch (Exception e) {
+            Log.w(TAG, "getAccessToken failed", e);
+            if (e instanceof UserRecoverableAuthException) {
```

Comment:
This is a nice finding indeed.
This will trigger UI during backup right? We have a mechanism to trigger the sign in UI if we throw AuthError like here: https://github.com/breez/breezmobile/pull/1085/files#diff-5826ba9b9fbe1aedc7cb899ee752912c591e4de18fb85c442a8f86dca296413eR60

Perhaps it is better only to throw here AuthError and let the app layer do the sign in job as it does today?

Reply:
Let me test if it works.

Reply:
@roeierez I just tried this https://gist.github.com/ademar111190/e0ed3c45350bd9a4a23589522dcb83c6 but it doesn't worked, am I missing something?